### ISSUE TITLE
Complex core dump error

### ISSUE THREAD

**Issue by TahaAslani** [refresh-bio/KMC#126]
Hi,

When I am trying to run complex command, I get this core dump error:
in1: 100% in2: 100% KMCCOMPLEXKMERLIST.sh: line 1: 14990 Segmentation fault      (core dumped) kmc_tools complex /home/taha/Desktop/AMR/KMCCOMPLEXLIST

Could you give me some ideas about how to work on debugging it?

Thanks,
Taha

**Comment by marekkokot** [comment]
Hi,

thanks for reporting that and for using kmc and kmc_tools. 
It would be great if I could reproduce your pipeline, are your input data publicly available? If yes could you give ma URL to download it, and your exact command lines and file with complex operation definition (KMCCOMPLEXLIST)?. Also, I suspect the number of input databases for kmc_tools complex is in your case 2, so probably you do not need to use complex but simple, though complex should also work of course, but you may check in the first step if kmc_tools works correctly when you use simple operation.


Regards,
Marek

**Comment by TahaAslani** [comment]
Hi Marek,

thanks a lot for your response. For some reason, I didn't see this earlier. I apologize for my late answer. I have uploaded my data to my google drive:
https://drive.google.com/drive/folders/1BGCEj_PMOxqnZwekJ7b6CN4F585ejYdl?usp=sharing
The command that I am using is:
kmc_tools complex MERGE
where MERGE is a text file.
kmc_tools works for other operations like dump.
For this mock example, I am merging 2 files. But for the project I have to do more as soon as I get the program working. 

I look forward to your response.

Many thanks,
Taha

**Comment by marekkokot** [comment]
Hi,

I tried to reproduce your crash, but it works on my machine. I have tried with the newest source codes, but I thought that maybe you are using one of the stable releases. 
I have checked KMC3.1.1 KMC3.1.0 and KMC3.0.0. The last one has failed, so the question is which KMC version you are using. If it is 3.0.0 than upgrading to the 3.1.1 should solve your problem. 
Let me know if it helps, if not we will discuss it further.

Best,
Marek
